jni_gen/lib/src/util/name_utils.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:jni_gen/src/elements/elements.dart';
+
+String getPackageName(String binaryName) => cutFromLast(binaryName, '.')[0];
+
+/// splits [str] into 2 from last occurence of [sep]
+List<String> cutFromLast(String str, String sep) {
```

Comment:
It's a shame that Dart doesn't have multi-return. Returning a List is pretty gross, but the only other option is writing a one-time-use class for the return type. A class would be cleaner, but I'll leave it up to you to decide if it's worth the effort.